src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useMemo } from 'react';
+import cn from 'classnames';
+import debounce from 'lodash.debounce';
+import { Person } from '../types/Person';
+
+export type Props = {
+  peopleFromServer: Person[],
+  setMyPerson: (person: Person) => void,
+  delay: number,
+};
+
+export const Autocomplete: React.FC<Props> = ({
+  peopleFromServer,
+  setMyPerson,
+  delay,
+}) => {
+  const [select, setSelect] = useState(false);
+  const [query, setQuery] = useState('');
+  const [chosePerson, setChosePerson] = useState('');
+
+  const applyQuery = useMemo(
+    () => debounce(setQuery, delay), [delay],
+  );
+
+  const filteredPeoples = useMemo(() => {
+    if (query !== '') {
+      return peopleFromServer.filter(
+        (person: Person) => person.name
+          .toLowerCase()
+          .includes(query.toLowerCase()),
+      );
+    }
+
+    return peopleFromServer;
+  }, [peopleFromServer, query]);
```

Comment:
let's make a code a little bit cleaner 
```suggestion
  const filteredPeoples = useMemo(() => {
    if (!query) {
        return peopleFromServer;
    }
    
    return peopleFromServer.filter(
       (person: Person) => person.name
          .toLowerCase()
          .includes(query.toLowerCase())
    );
  }, [peopleFromServer, query]);
```